universe/ObjectMap.cpp

```diff
@@ -225,23 +235,30 @@ std::shared_ptr<UniverseObject> ObjectMap::Remove(int id) {
     // and erase from pointer maps
     m_objects.erase(it);
     FOR_EACH_SPECIALIZED_MAP(EraseFromMap, id);
-    m_existing_objects.erase(id);
-    m_existing_buildings.erase(id);
-    m_existing_fields.erase(id);
-    m_existing_fleets.erase(id);
-    m_existing_ships.erase(id);
-    m_existing_planets.erase(id);
-    m_existing_pop_centers.erase(id);
-    m_existing_resource_centers.erase(id);
-    m_existing_systems.erase(id);
+    FOR_EACH_EXISTING_MAP(EraseFromMap, id);
     return result;
 }
 
-void ObjectMap::Clear()
-{ FOR_EACH_MAP(ClearMap); }
+void ObjectMap::Clear() {
+    FOR_EACH_MAP(ClearMap);
+    FOR_EACH_EXISTING_MAP(ClearMap);
+}
 
 void ObjectMap::swap(ObjectMap& rhs)
-{ FOR_EACH_MAP(SwapMap, rhs); }
+{
+    // ObjectMap doesn't have access to maps of existing objects based on type
```

Comment:
explain / clarify this comment?

Reply:
It's not clear why that's relevant, as `ObjectMap::Map<T>` is not (explicitly?) used here.

Reply:
`SwapMap` uses `ObjectMap::Map<T>` to determine which map to use.

Reply:
Where is `SwapMap` defined?

Reply:
Fixed.